### ISSUE TITLE
Transaction.del  error: ambiguous reference to overloaded definition

### ISSUE THREAD

**Issue by 123avi** [xetorthio/jedis#566]
Using Scala 2.10.x and Transaction.del ('redis.clients:jedis:2.4.0') yields 
error: ambiguous reference to overloaded definition
both method del in class MultiKeyPipelineBase of type (x$1: <repeated...>[String])
redis.clients.jedis.Response[Long]
and  method del in class PipelineBase of type (x$1: String )
redis.clients.jedis.Response[Long]
match argument types (String)

**Comment by matthieu-ravey** [comment]
Is there any update on this issue ?

**Comment by marcosnils** [comment]
@matthieu-ravey did you run into this issue?. Which Jedis version are you using?

**Comment by matthieu-ravey** [comment]
Yes I just had this issue. I'm using Jedis 2.6.2 and Scala 2.11

**Comment by marcosnils** [comment]
@matthieu-ravey hmm none of the main Jedis contributors actually uses Scala so I'm not sure where the problem might be. For the exception seems like it was nothing to do with Jedis internally but rather a Jedis/Java - Scala binding. Honestly, I have no immediate idea about how to debug this.

**Comment by HeartSaVioR** [comment]
I'm learning Scala now, so seems like I may take care of it for now. :)

**Comment by HeartSaVioR** [comment]
I found the issue.

RedisPipeline has ```Response<Long> del(String key)```, and MultiKeyCommandsPipeline has  
```Response<Long> del(String... keys)```.

It's not just a del method issue. Seems like varargs made a problem. 
I think Scala's complaining is not wrong so we should clarify signature.

tl; dr
Overloaded methods which one is one arg and other one is vararg are conflicted.

Easiest way to resolve is remove ```Response<Long> del(String key)``` from RedisPipeline but we cannot use del from ShardedJedis so it doesn't make sense.
Other way to resolve is removing usage of varargs, and replace it to List. 
Maybe we can make conversion util method which converts T...args to List<T> though it's not convenient compared to pure varargs.

@xetorthio @marcosnils @nykolaslima What do you think?

**Comment by marcosnils** [comment]
@HeartSaVioR what if we create another `del(List<String> keys)` method just to make this work in Scala?, Will that make this case work?

I quickly  checked all the other methods and I believe that this is the only one that's causing problems. 

I don't like either the idea of removing the `del (String... keys)` method because as you said it won't work for ShardedJedis or JedisCluster; and replace all the varargs to List<T> also doesn't seem like a good option. 

What do you think?

**Comment by HeartSaVioR** [comment]
@marcosnils 
Sorry to respond later.

First of all, problem will be not solved when we add del(List<String> keys), and call del("a").
In order to avoid it Scala users should always call del(List("a")) though IDE shows there's del(String key) and del(String... keys). It doesn't make sense.

I've tested some and found a solution which would make us annoying (and may you know already), but I think it would be only way to support both single arg and varargs.

* Solution: change del(String... keys) to del(String firstKey, String secondKey, String... remainKeys)

If we're OK to change, then we should check all operations which have methods that single arg and varargs.

Does it make sense?

cc. @123avi @matthieu-ravey @xetorthio @nykolaslima

**Comment by s-boyle** [comment]
Here is how I solved it....

Broken code:
j.del("mystring")

Working code:
val stringArray = Array[String]("mystring")
j.del(stringArray:_*)

I believe that the _* turns the array into a varargs format

**Comment by sazzad16** [comment]
Refer to #1263 
Until then, use workaround.
